controllers/component_build_controller_pac.go

```diff
@@ -707,6 +765,13 @@ func (r *ComponentBuildReconciler) ensurePaCRepository(ctx context.Context, comp
 		return err
 	}
 
+	ns, err := r.getNamespace(ctx, component.GetNamespace())
+	if err != nil {
+		log.Error(err, "failed to get namespace where component presents for setting custom parameter.")
+		return err
+	}
+	pacRepoAddCustomParams(repository, ns)
+
 	existingRepository := &pacv1alpha1.Repository{}
```

Comment:
Does this work correctly with regards to the `existingRepository` handling? It looks like our code will only `Create` a repo if it doesn't exist, but won't `Update` one if it does. Shouldn't we also add the custom param to repositories that already exist?

Reply:
This implementation is for a new Repository CR. `ensurePaCRepository` is called during the PaC provision. The existing Repository CRs can't be updated until users reprovision the PaC (i.e. re-enable custom build).

Reply:
We should think about migration. It could be a script that we provide to SREs